### DISCUSSION TITLE
Variable to differentiate Between build time and revalidation

### DISCUSSION THREAD

**Discussion by adityarao04** [vercel/next.js#42890] [Help]
The getStaticProps runs on buildtime as well as during revalidation. Is there some sort of identifier to differentiate between build time and revalidation. I need to perform some different types of actions in getSaticProps during build time and runtime. can someone help me with this?

**Comment by icyJoseph** [comment]
During build time there's a env var, called `NEXT_PHASE`, which you can read and then compare like this:

```ts
import { PHASE_PRODUCTION_BUILD } from "next/constants";

// later on
export const getStaticProps = ({ preview, revalidate }) => {
 const isBuildStep = process.env.NEXT_PHASE === PHASE_PRODUCTION_BUILD
```

Read more on this answer I gave a while ago, but nobody really confirmed it worked for them: https://github.com/vercel/next.js/discussions/42673#discussioncomment-4094721
